core/ledger/kvledger/channelinfo_provider.go

```diff
@@ -0,0 +1,79 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package kvledger
+
+import (
+	cb "github.com/hyperledger/fabric-protos-go/common"
+	"github.com/hyperledger/fabric/bccsp/factory"
+	"github.com/hyperledger/fabric/common/channelconfig"
+	"github.com/hyperledger/fabric/common/ledger/blkstorage"
+	"github.com/hyperledger/fabric/protoutil"
+)
+
+// ChannelInfoProvider implements ledger.ChannelInfoProvider interface
+type ChannelInfoProvider struct {
+	BlkStoreProvider *blkstorage.BlockStoreProvider
+}
```

Comment:
When will this be used? Will the `kv_ledger_provider.go` create a new `ChannelInfoProvider` and pass it to privacyenabledstate.`GetDBHandle()` while opening the ledger for a given channel so that it can call the `DeployedInforCCProvider`? 

If this struct is to be created by the kvledger pkg, shouldn't we have the following?
```suggestion
type ChannelInfoProvider struct {
	blockStore *blkstorage.BlockStore
}

func NewChannelInfoProvider(channelName string) (*ChannelInfoProvider, error) {
        blockStore, err := p.BlkStoreProvider.Open(channelName)
	if err != nil {
		return nil, err
	}
	return &ChannelInfoProvider{blockStore}, err
}
```
**OR**
```suggestion
type ChannelInfoProvider struct {
	blockStore *blkstorage.BlockStore
}

func NewChannelInfoProvider(blockStore blkstorage.BlockStore) (*ChannelInfoProvider, error) {
	return &ChannelInfoProvider{blockStore}, err
}
```